packages/commands/js-debugger/src/adapters/browser-adapter.ts

```
@@ -0,0 +1,717 @@
+      stackTrace: params.stackTrace
+        ? this.parseStackTrace(params.stackTrace)
+        : undefined,
+    };
+
+    this.consoleHandlers.forEach((handler) => {
+      try {
+        handler(message);
+      } catch (error) {
+        console.warn('Error in console handler:', error);
+      }
+    });
+  }
+
+  /**
+   * Handle runtime exception
+   */
+  private handleException(params: CDPExceptionThrownParams): void {
+    const message: ConsoleMessage = {
+      level: 'error',
+      timestamp: new Date().toISOString(),
+      message:
+        params.exceptionDetails.exception?.description ||
+        params.exceptionDetails.text,
+      args: [params.exceptionDetails.exception?.value],
+      stackTrace: params.exceptionDetails.stackTrace
+        ? this.parseStackTrace(params.exceptionDetails.stackTrace)
+        : undefined,
+    };
+
+    this.consoleHandlers.forEach((handler) => {
+      try {
+        handler(message);
+      } catch (error) {
+        console.warn('Error in console handler:', error);
+      }
+    });
+  }
+
+  /**
+   * Load source map for a script
+   */
+  private async loadSourceMap(
+    scriptId: string,
+    sourceMapURL: string,
+  ): Promise<void> {
+    try {
+      // Handle relative URLs and data URLs
+      let sourceMapContent: string;
+
+      if (sourceMapURL.startsWith('data:')) {
+        // Data URL
+        const base64Data = sourceMapURL.split(',')[1];
+        sourceMapContent = Buffer.from(base64Data, 'base64').toString('utf-8');
+      } else if (sourceMapURL.startsWith('http')) {
+        // Absolute URL - try to use built-in fetch or skip
+        try {
+          // Use Node.js built-in fetch (Node 18+) if available
+          if (typeof globalThis.fetch === 'function') {
+            const response = await globalThis.fetch(sourceMapURL);
+            sourceMapContent = await response.text();
+          } else {
+            // Skip HTTP source maps if fetch is not available
```

Minor: When fetch is not available, source map loading silently skips with only a console.warn. Consider providing a more explicit mechanism to inform users about degraded debugging experience, or document this limitation in the browser adapter documentation.